iaf-commons/src/main/java/nl/nn/adapterframework/util/StringDataSource.java

```diff
@@ -0,0 +1,41 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.util;
+
+import java.io.IOException;
+
+public interface StringDataSource {
```

Comment:
Ik denk wel dat ik snap waar je naar toe wilt maar misschien dit nog even achterwege laten.

Reply:
> Ik denk wel dat ik snap waar je naar toe wilt maar misschien dit nog even achterwege laten.

Maar dit heb ik nodig om de afhankelijkheid op `Message` uit verschillende andere classes te kunnen halen.

Reply:
De methode StreamUtil#getOutputStream kan naar JdbcUtil toe, dan is daar iig de afhankelijkheid weg.
De StringResolver heeft het helaas nodig omdat in de FixedResultPipe de PipeLineSession mee wordt gegeven.
Nu zit er wel al een check in op Properties, maar dat is de PipeLineSession ook niet echt... Lastig.
Om het echt mooi op te lossen zou je een proxy class kunnen maken die properties extend, en dan daar de getProperty overwriten...

Reply:
Ik zou ook in Core een `AdditionalStringResolver` implementatie kunnen toevoegen die kijkt of een property bij een key een `Message` is, en dan de `StringDataSource` weg kunnen halen?

Vergelijkbaar hoe ik dat heb opgelost voor de CredentialFactory.